packages/ignite-core/tasks/installation.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+const readline = require("readline");
+const xsh = require("xsh");
+const logger = require("../lib/logger");
+const chalk = require("chalk");
+const errorHandler = require("../lib/error-handler");
+const semverComp = require("../lib/semver-comp");
+const igniteCore = require("../ignite");
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  terminal: false
+});
+
+const installXClapCLI = function(type, igniteCore) {
+  return rl.question("Proceed? (y/n) ", answer => {
+    if (answer.toLowerCase() === "y") {
+      return xsh
+        .exec("npm install -g xclap-cli")
+        .then(function() {
+          logger.log(
+            chalk.green("You've successfully installed the latest xclap-cli.")
+          );
+
+          if (type && igniteCore) {
+            logger.log(chalk.green("Please choose your next task:"));
+            return igniteCore(type);
+          }
+
+          return ;
+        })
+        .catch(err =>
+          errorHandler(err, "Failed at: Installing the latest xclap-cli.")
+        );
+    }
+  });
+};
+
+const checkXClapCLI = function() {
+  return new Promise((resolve, reject) => {
+    return xsh
+      .exec(true, "npm ls -g -j --depth=0 xclap-cli")
+      .then(function(ret) {
+        resolve(JSON.parse(ret.stdout).dependencies["xclap-cli"].version);
```

Comment:
`xsh.exec` returns a promise.  You don't need to create another promise like this.